docs/reference/core/zioapp.md

```diff
@@ -109,3 +109,32 @@ object Main extends ZIOApp.Proxy(MyApp1 <> MyApp2)
 ```
 
 The `<>` operator combines the layers of the two applications and then runs the two applications in parallel.
+
+## Graceful Shutdown Timeout
+
+When a ZIO application (e.g. one extending `ZIOAppDefault`) receives an external interruption signal such as **SIGINT** when pressing **Ctrl+C**, the runtime will attempt to run all finalizers (cleanup logic) before exiting. By default, `gracefulShutdownTimeout` returns `Duration.Infinity`, which means ZIO will wait forever for finalizers unless you override.
+
+Override `gracefulShutdownTimeout` to bound how long the runtime should wait for finalizers. For example, to wait at most 30 seconds:
+
+```scala mdoc:compile-only
+import zio._
+
+object MyApp extends ZIOAppDefault {
+  // Wait at most 30 seconds for all finalizers to complete on SIGINT
+  override def gracefulShutdownTimeout: Duration = Duration.fromSeconds(30)
+
+  val run =
+    ZIO.logInfo("MyApp is running...") *>
+    ZIO.never
+}
```

Comment:
Please provide an example that has finalizers. Also provide one that has finalizers that where the timeout will be breached